File: templates/problems/17_math_geometry/04_happy_number/happy_number_test.go
package happy_number

import "testing"

func TestIsHappy(t *testing.T) {
	tests := []struct {
		name     string
		n        int
		expected bool
	}{
		{"happy 19", 19, true},
		{"happy 1", 1, true},
		{"not happy 2", 2, false},
		{"happy 7", 7, true},
		{"not happy 4", 4, false},
		{"happy 100", 100, true},
		{"not happy 20", 20, false},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := isHappy(tt.n)
			if got != tt.expected {
				t.Errorf("isHappy(%d) = %v, want %v", tt.n, got, tt.expected)
			}
		})
	}
}

File: templates/problems/17_math_geometry/04_happy_number/typescript/happyNumber.test.ts
import { describe, it, expect } from 'vitest';
import { isHappy } from './happyNumber';

describe('HappyNumber', () => {
    it('happy 19', () => {
        expect(isHappy(19)).toBe(true);
    });

    it('happy 1', () => {
        expect(isHappy(1)).toBe(true);
    });

    it('not happy 2', () => {
        expect(isHappy(2)).toBe(false);
    });

    it('happy 7', () => {
        expect(isHappy(7)).toBe(true);
    });

    it('not happy 4', () => {
        expect(isHappy(4)).toBe(false);
    });

    it('happy 100', () => {
        expect(isHappy(100)).toBe(true);
    });

    it('not happy 20', () => {
        expect(isHappy(20)).toBe(false);
    });
});
